.github/workflows/hub-upload.yml

```diff
@@ -1,24 +1,24 @@
 Name: Upload pod
-on:
-  repository_dispatch:
-  types: pod-upload
+on: [repository_dispatch]
 jobs:
   build:
     name: Dockerize pod and upload with repo dispatch
     runs-on: ubuntu-latest
     steps:
       - name: 'Cloning user repo'
+        if: github.event.action == 'pod-upload'
         uses: actions/checkout@v2
         with:
           token: ${{ secrets.JINA_DEV_BOT }}
       - env:	
           GITHUB_TOKEN: ${{ secrets.JINA_DEV_BOT }}
-          repository: ${{ clone_repo_url }}
+          repository: ${{ github.event.client_payload.repository }}
           path: jinaai-jina
       - name: Jina Hub Image Builder and Push to docker hub
+        if: github.event.action == 'pod-upload'
```

Comment:
instead of having  `if: github.event.action == 'pod-upload'` in 2 places, please move this to L#5